### PR TITLE
feat: add support for API_PORT to environment

### LINKED ISSUE #96: Sync env vars for gateway
As a developer, I want clear documentation regarding the available options for configuring a service via the environment. When multiple related services each have environment variables that service the same purpose, they should be consistently named. A sample .env.template file or README documentation should document all available environment variables, their purpose, range of values, and whether required or optional.

- [x] Honor `API_PORT` for binding app listener interface
- [x] Ensure all variables are properly documented in:
    - [x] ENVIRONMENT.md
    - [x] .env.template

### PR DESCRIPTION
**Pull request by JoeCap08055**
This PR does the following:
* Adds support for a standard `API_PORT` environment variable, and updates documentation surrounding the environment.
* Updates/adds documentation surrounding environment variables
* Updates `env.template` with sane defaults & comments for all environment variables 

Closes #96

### CODE REVIEW

**Review by saraswatpuneet — APPROVED** [review]

File: ENVIRONMENT.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+| `FREQUENCY_URL`                           | Blockchain node address                                                                                                                                                                                                  |      http(s): or ws(s): URL      |      Y       |         |
+| `GRAPH_ENVIRONMENT_DEV_CONFIG`            | [NOTE]: Only used for Dev environments. Add the graph environment config in JSON format only used for Dev environments. Be careful to escape any inner quotes as this is in a .env file and will be parsed by the shell. | JSON [(example)](./env.template) | Y (if 'Dev') |         |
+| `GRAPH_ENVIRONMENT_TYPE`                  | Graph environment type. This can be 'Dev' or 'Rococo' or 'Mainnet'.                                                                                                                                                      |       Dev\|Rococo\|Mainnet       |      Y       |         |
```
Comment (wilwade) [inline ENVIRONMENT.md:13]: Is there something we need to do to support `Paseo`?

File: ENVIRONMENT.md
Comment (wilwade) [inline ENVIRONMENT.md:1]: We should likely link this from the readme somewhere

File: ENVIRONMENT.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+| `PROVIDER_ACCOUNT_SEED_PHRASE`            | Seed phrase for provider MSA control key                                                                                                                                                                                 |              string              |      Y       |         |
+| `PROVIDER_BASE_URL`                       | Base URL for provider webhook endpoints                                                                                                                                                                                  |               URL                |      Y       |         |
+| `PROVIDER_ID`                             | Provider MSA ID                                                                                                                                                                                                          |             integer              |      Y       |         |
```
Comment (wilwade) [inline ENVIRONMENT.md:20]: Do we do a initialization check to make sure that our Provider Id is correct? If not, we should add that as a todo for this and other gateway services with a key. It is a good double check that we are connecting to the correct network.

File: env.template [hunk 3]
```diff
@@ -1,25 +1,62 @@
...
+HEALTH_CHECK_MAX_RETRY_INTERVAL_SECONDS=10
+
+# Minimum number of consecutive successful calls to the provider webhook
+# `/health` endpoint before it is marked up again
+HEALTH_CHECK_SUCCESS_THRESHOLD=10
+
+# Maximum amount of provider capacity this app is allowed to use (per epoch)
+#     type: 'percentage' | 'amount'
+#     value: number (may be percentage, ie '80', or absolute amount of capacity)
 CAPACITY_LIMIT='{"type":"percentage", "value":80}'
```
Comment (wilwade) [inline env.template:54]: I found at least for me, I had to remove the `'` marks:

```suggestion
CAPACITY_LIMIT={"type":"percentage", "value":80}
```

File: env.template [hunk 4]
```diff
@@ -1,25 +1,62 @@
...
+# How many minutes to delay between successive scans of the chain
+# for new accounts (after end of chain is reached)
 BLOCKCHAIN_SCAN_INTERVAL_MINUTES=1
+
+# Max number of jobs allowed on the 'graphUpdateQueue' before
+# blockchain scan will be paused to allow queue to drain
 QUEUE_HIGH_WATER=1000
+
+# Seed phrase for provider MSA control key
 PROVIDER_ACCOUNT_SEED_PHRASE='come finish flower cinnamon blame year glad tank domain hunt release fatigue'
```
Comment (wilwade) [inline env.template:33]: Same here. I had to remove the quotes
```suggestion
PROVIDER_ACCOUNT_SEED_PHRASE=come finish flower cinnamon blame year glad tank domain hunt release fatigue
```

File: env.template [hunk 5]
```diff
@@ -1,25 +1,62 @@
...
 FREQUENCY_TX_TIMEOUT_SECONDS=60
-# An optional bearer token may be specified for provider authentication
-PROVIDER_ACCESS_TOKEN=some-token
-DEAD_LETTER_JOB_PREFIX=dead-letter-job
 
-# Add the graph environment type. This can be 'Dev' or 'Rococo' or 'Mainnet'.
+# Graph environment type. This can be 'Dev' or 'Rococo' or 'Mainnet'.
 GRAPH_ENVIRONMENT_TYPE=Dev
 
 # [NOTE]: The following config is only used for Dev environments.
```
Comment (wilwade) [inline env.template:62]: I had to remove the `'` marks for the `GRAPH_ENVIRONMENT_DEV_CONFIG` as well

**Review by wilwade — COMMENTED** [review]
A few notes to review, but nothing super blocking.

File: ENVIRONMENT.md (hunk 1 above)
Comment (JoeCap08055) [reply ENVIRONMENT.md:13]: Paseo would probably use the mainnet config for now, actually--the schema ids on paseo match mainnet, due to how they were deployed.

Comment (JoeCap08055) [reply ENVIRONMENT.md:13]: I'll file a ticket in `graph-sdk` to put in a Paseo config.

File: ENVIRONMENT.md (hunk 2 above)
Comment (JoeCap08055) [reply ENVIRONMENT.md:20]: No, we don't... I'll add a ticket

File: env.template (hunk 3 above)
Comment (wilwade) [reply env.template:54]: Now that I look closer, it was all inside of Docker the issues happened. So it was the `.env.docker.dev` where the issue was occurring.

File: ENVIRONMENT.md
Comment (JoeCap08055) [reply ENVIRONMENT.md:1]: Done.

File: ENVIRONMENT.md (hunk 1 above)
Comment (JoeCap08055) [reply ENVIRONMENT.md:13]: nvm; `graph-sdk` already has a config for `TestnetPaseo`. I'll update the docs to include that option.

**Pull request merged**